celery/app/base.py

```diff
@@ -342,6 +344,30 @@ def close(self):
         self._pool = None
         _deregister_app(self)
 
+    def start(self, argv=None):
+        from celery.bin.celery import celery
+
+        celery.params[0].default = self
+
+        try:
+            celery.main(args=argv, standalone_mode=False)
+        except Exit as e:
+            return e.exit_code
+        finally:
+            celery.params[0].default = None
+
+    def worker_main(self, argv=None):
+        if argv is None:
+            argv = ['worker']
```

Comment:
This used to be documented such that omitting `argv` would end up using `sys.argv`. I assume that if we don't pass an `argv` down to `celery.main()`, `click` will end up doing that for us. But here, IIUC, we'll call `self.start(['worker'])` which won't give `click` the chance to notice that the user didn't pass in an `argv` and fall back to using `sys.argv`.

I figure we could either change this to
```suggestion
        if argv is None:
            argv = sys.argv
```
and still perform the check for the worker subcommand below. Or if we could somehow tell click that the only valid subcommand is now `worker`, that might be an interesting approach. No idea if that's a thing click would allow/make easy though.